metadata/snapshot.go

```diff
@@ -223,6 +223,10 @@ func (s *snapshotter) Update(ctx context.Context, info snapshots.Info, fieldpath
 		return snapshots.Info{}, err
 	}
 
+	if _, err := s.Snapshotter.Update(ctx, info, fieldpaths...); err != nil {
```

Comment:
I prefer to move it into `update` function in a transaction. :)

and according to `prepare` function, the option will be filtered before passing it to snapshotter. I think we should make it consistent. 

ref: https://github.com/containerd/containerd/pull/3055

Reply:
Agreed, this must be performed inside the transaction to keep consistency. With the filtered labels.

Reply:
Done